VideoComponent.cs

```diff
@@ -28,64 +32,87 @@ private class VLCErrorException : Exception { }
 
         public override float MinimumWidth => 10;
 
-        public AxVLCPlugin2 VLC { get; set; }
-        public bool Initialized { get; set; }
+        private static LibVLC libVLC;
 
-        public VideoComponent(LiveSplitState state)
-            : this(state, CreateVLCControl())
-        { }
+        public MediaPlayer mediaPlayer => ((VideoView)Control).MediaPlayer;
 
-        public VideoComponent(LiveSplitState state, AxVLCPlugin2 vlc)
-            : base(state, vlc, ex => ErrorCallback(state.Form, ex))
+        static VideoComponent()
+        {
+            Core.Initialize();
+            libVLC = new LibVLC(new string[] { });
+        }
+
+        public VideoComponent(LiveSplitState state) : this(state, CreateVLCControl())
+        {
+        }
+
+        public VideoComponent(LiveSplitState state, VideoView vview) : base(state, vview, ex => ErrorCallback(state.Form, ex))
         {
             Settings = new VideoSettings();
             State = state;
-            VLC = vlc;
+            //set up the MediaPlayer BEFORE adding it to the VideoView
+            var mediaplayer = new MediaPlayer(libVLC)
+            {
+                EnableKeyInput = false,
+                EnableMouseInput = false,
+                EnableHardwareDecoding = false
+            };
+            vview.MediaPlayer = mediaplayer;
 
             state.OnReset += state_OnReset;
             state.OnStart += state_OnStart;
             state.OnPause += state_OnPause;
             state.OnResume += state_OnResume;
         }
 
+        public delegate void InvokeAction(ISynchronizeInvoke control, Action action);
+
+        public static void Lock(ISynchronizeInvoke control, Action action)
+        {
+            lock (control)
+            {
+                action();
+            }
+        }
+
+        public new void InvokeIfNeeded(Action x)
+        {
+            if (Control != null && Control.InvokeRequired)
+                Control.Invoke((InvokeAction)Lock, Control, x);
+            else
+                Lock(Control,x);
+        }
+
         static void ErrorCallback(Form form, Exception ex)
         {
-            string requiredBits = Environment.Is64BitProcess ? "64" : "32";
-            MessageBox.Show(form, "VLC Media Player 2.2.1 (" + requiredBits + "-bit) along with the ActiveX Plugin need to be installed for the Video Component to work.", "Video Component Could Not Be Loaded", MessageBoxButtons.OK, MessageBoxIcon.Error);
+            //string requiredBits = Environment.Is64BitProcess ? "64" : "32";
```

Comment:
Could you get rid of all the commented out code introduced in this PR, such as this?

Reply:
Yes. Also I just realized this error message no longer applies since we're not relying on ActiveX anymore.